pcweb/flexdown.py

```diff
@@ -0,0 +1,147 @@
+
+
+custom_styles = {
+    "h1": {
+        "as_": "h1",
+        "size": "xl",
+        "color": styles.text_colors["docs"]["header"],
+        "font_weight": styles.font_weights["heading"],
+        "font_family": styles.SANS,
+    },
+    "h2": {
+        "as_": "h2",
+        "font_size": styles.H3_FONT_SIZE,
+        "margin_top": "1em",
+        "color": styles.text_colors["docs"]["header"],
+        "font_weight": styles.font_weights["subheading"],
+        "font_family": styles.SANS,
+    },
+    "a": styles.LINK_STYLE,
+    "p": {
+        "as_": "p",
+        "margin_y": "1em",
+        "font_family": styles.SANS,
+    },
+    "code": {
+        "color": "#1F1944",
+        "bg": "#EAE4FD",
+    },
+    "pre": {
+        "border_radius": styles.DOC_BORDER_RADIUS,
+        "background": "transparent",
+        "code_tag_props": {
+            "style": {
+                "fontFamily": "inherit"
+            }
+        }
+    }
+}
+
+def md(source: str):
+    return rx.markdown(
+        source,
+        custom_styles=custom_styles,
+    )
+
+def parse_markdown_front_matter(markdown_content):
+    # Define the regular expression pattern to match front matter
+    pattern = r"^---\s*\n(.+?)\n---\s*\n(.*)$"
+    # Extract the front matter and content using the pattern
+    match = re.match(pattern, markdown_content, re.DOTALL)
+    if match:
+        front_matter = match.group(1)
+        content = match.group(2)
+        # Parse the front matter as YAML
+        try:
+            front_matter_data = yaml.safe_load(str(front_matter))
+        except yaml.scanner.ScannerError:
+            front_matter_data = front_matter
+
+        if isinstance(front_matter_data, str):
+            front_matter_data = front_matter
+
+        return front_matter_data, content
+    else:
+        return None, markdown_content
+
+def evaluate_template_string(input_string, scope):
+    # Regular expression to match the template placeholders
+    template_regex = r"{(.*?)}"
+    matches = re.findall(template_regex, input_string)
+
+    for match in matches:
+        try:
+            # Evaluate the Python expression and replace the template placeholder
+            eval_result = str(eval(match, scope))
+            input_string = input_string.replace("{" + match + "}", eval_result)
+        except Exception as e:
+            # If the evaluation fails, leave the template placeholder unchanged
+            print(f"Failed to evaluate expression '{match}': {e}")
+
+    return input_string
+
+def parse(source: str, md=md):
+    """Parse out code blocks annotated with ```reflex
+    and replace them with the output of the code block.
+    The surrounding Markdown should be left untouched.
+    """
+    front_matter, source = parse_markdown_front_matter(source)
+    if isinstance(front_matter, str):
+        exec(front_matter)
+
+    lines = source.split("\n")
+    output = []
+    in_reflex_block = False
+    current_block = []
+
+    for line in lines:
```

Comment:
This appears to cause our python code in markdown to break if we have vertical spaces in the code:

```python
import reflex as rx


config = rx.Config(
    app_name="hello",
)
```
Here the code is not all in the code box, it stops at config = rx.Config( app_name="hello", )

Reply:
Got it, will add this also in my follow up.